src/visitor.rs

```diff
@@ -132,22 +133,44 @@ impl<'a> FmtVisitor<'a> {
         self.buffer.push_str("{");
 
         if self.config.remove_blank_lines_at_start_or_end_of_block() {
-            if let Some(stmt) = b.stmts.first() {
-                let snippet = self.snippet(mk_sp(self.last_pos, stmt.span.lo));
-                let len = CommentCodeSlices::new(&snippet)
-                    .nth(0)
-                    .and_then(|(kind, _, s)| {
-                        if kind == CodeCharKind::Normal {
-                            // There may be inner attributes
-                            let s = &s[..s.len() -
-                                           s.trim_left_matches(&[' ', '\t', '\r', '\n'][..]).len()];
-                            s.rfind('\n')
+            if let Some(first_stmt) = b.stmts.first() {
+                let attr_lo = inner_attrs
+                    .and_then(|attrs| {
+                        attrs
+                            .iter()
+                            .filter(|a| a.style == ast::AttrStyle::Inner)
+                            .nth(0)
+                            .map(|attr| attr.span.lo)
+                    })
+                    .or_else(|| {
+                        // syntax BUG: `attrs()` returns nothing for items
```

Comment:
It turned out that the behavior of `attrs()` is intentional; attributes for an item in a statement position don't belong to the statement (https://github.com/rust-lang/rust/commit/b968ee36564abd96da72c381f49bf8009f4b1870). Perhaps `Spanned` follows the convention.